data/store/mongo/mongo_data.go

```diff
@@ -1015,50 +989,64 @@ func (d *DataRepository) GetLastUpdatedForUser(ctx context.Context, id string) (
 		return nil, nil
 	}
 
-	status.LastUpload, err = time.Parse(time.RFC3339Nano, *dataSet[0].CreatedTime)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to parse latest CreatedTime")
-	}
+	status.LastUpload = *dataSet[0].CreatedTime
 	status.LastUpload = status.LastUpload.UTC()
 
-	status.LastData, err = time.Parse(time.RFC3339Nano, *dataSet[0].Time)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to parse latest Time")
-	}
+	status.LastData = *dataSet[0].Time
 	status.LastData = status.LastData.UTC()
 
-	return &status, nil
+	return status, nil
 }
 
-func (d *DataRepository) DistinctCGMUserIDs(ctx context.Context) ([]string, error) {
-	var userIDs []string
+func (d *DataRepository) DistinctUserIDs(ctx context.Context) ([]string, error) {
+	var distinctUserIDMap = make(map[string]struct{})
+	var empty struct{}
 
 	if ctx == nil {
-		return userIDs, errors.New("context is missing")
+		return nil, errors.New("context is missing")
 	}
 
 	// allow for a small margin on the pastCutoff to allow for calculation delay
 	pastCutoff := time.Now().AddDate(0, -23, -20).UTC()
 	futureCutoff := time.Now().AddDate(0, 0, 1).UTC()
 
-	// we don't query for users with different time field types, as users with the new types would
-	// not exist before summaries were launched.
-	selector := bson.M{
+	selectorOld := bson.M{
+		"_userId": bson.M{"$ne": -1111},
+		"_active": true,
+		"type":    bson.M{"$ne": -1111},
 		"time": bson.M{"$gte": pastCutoff.Format(time.RFC3339Nano),
 			"$lte": futureCutoff.Format(time.RFC3339Nano)},
-		"_active": true,
-		"type":    "cbg",
+	}
+
+	selector := bson.M{
 		"_userId": bson.M{"$ne": -1111},
+		"_active": true,
+		"type":    bson.M{"$ne": -1111},
+		"time":    bson.M{"$gte": pastCutoff, "$lte": futureCutoff},
 	}
 
-	// we would prefer to use hints here instead of _userId $ne -1111, but hints are not (yet?) available on distinct opts
 	result, err := d.Distinct(ctx, "_userId", selector)
 	if err != nil {
-		return userIDs, errors.Wrap(err, "error fetching distinct userIDs")
+		return nil, errors.Wrap(err, "error fetching distinct userIDs")
+	}
+
+	resultOld, err := d.Distinct(ctx, "_userId", selectorOld)
+	if err != nil {
+		return nil, errors.Wrap(err, "error fetching distinct userIDs")
 	}
 
 	for _, v := range result {
-		userIDs = append(userIDs, v.(string))
+		distinctUserIDMap[v.(string)] = empty
+	}
+	for _, v := range resultOld {
+		distinctUserIDMap[v.(string)] = empty
+	}
+
+	userIDs := make([]string, len(distinctUserIDMap))
+	offset := 0
```

Comment:
nitpick: Could do a `userIDs := make([]string, 0, len(distinctUserIDMap))` and then `userIDs = append(userIDs, k)` later. Would remove the need for temporary `offset` variable

Reply:
i had to read up on exactly append works on a capacity specified array, and it appears you're right, i thought append was always a copy, and you had to write to the index to not have it copy, some other services use the same pattern (a temporary offset).

Switched this out for your way, TIL.